src/rdkafka_queue.c

```diff
@@ -587,6 +614,10 @@ int rd_kafka_q_serve_rkmessages (rd_kafka_q_t *rkq, int timeout_ms,
                                        RD_KAFKA_Q_CB_RETURN, NULL);
                 if (res == RD_KAFKA_OP_RES_KEEP ||
                     res == RD_KAFKA_OP_RES_HANDLED) {
+                        if (!rko && unlikely(rko->rko_type == RD_KAFKA_OP_BARRIER)) {
```

Comment:
This if-statement is wrong, both checking for NULL and then derferencing that null pointer.
But there is no case where `rko` is NULL here though, so this code block is never reached.
If the new tests pass with this (faulty) code it means the tests don't really cover what it is supposed to test.